exp/runtime/api/v1beta2/extensionconfig_types.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta2"
+)
+
+// ANCHOR: ExtensionConfigSpec
+
+// ExtensionConfigSpec defines the desired state of ExtensionConfig.
+type ExtensionConfigSpec struct {
+	// clientConfig defines how to communicate with the Extension server.
+	// +required
+	ClientConfig ClientConfig `json:"clientConfig"`
+
+	// namespaceSelector decides whether to call the hook for an object based
+	// on whether the namespace for that object matches the selector.
+	// Defaults to the empty LabelSelector, which matches all objects.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// settings defines key value pairs to be passed to all calls
+	// to all supported RuntimeExtensions.
+	// Note: Settings can be overridden on the ClusterClass.
+	// +optional
+	Settings map[string]string `json:"settings,omitempty"`
+}
+
+// ClientConfig contains the information to make a client
+// connection with an Extension server.
+type ClientConfig struct {
```

Comment:
ClientConfig is required but does not have any required fields within it, so I could just created something with
```
clientConfig: {}
```
What would that mean? Is that valid? Should this actually be optional/have a required field/have a `MinProperties`?

Reply:
As long as this is the only potential change (?), we could consider doing it here. Otherwise we should add it to:

- #10852

@sbueringer @fabriziopandini

Reply:
I think lets track it and tackle separately. We could probably do a more thorough API review of these types to be honest

Reply:
A lot of this might be aligned to Mutating?ValidatingWebhookConfiguration

Reply:
@chrischdi Feel free to add a follow-up to the umbrella issue